docker/docker.go

```diff
@@ -264,6 +265,31 @@ func WithSharedDirectory(hostPath, containerPath string) ConfigOption {
 }
 
 func withVolume(typ mount.Type, hostPath, containerPath string) ConfigOption {
+	c, err := client.NewEnvClient()
+	if err != nil {
+		logrus.Fatalf("could not create docker client")
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+
+	info, err := c.Info(ctx)
+	if err != nil {
+		logrus.Fatalf("could not get information")
+	}
+
+	isWinHost := info.OSType == "windows" || strings.Contains(
```

Comment:
Initially, I thought that the first check was enough. But for example in the Windows Server we're using the "OSType" is actually equal to "linux" and the "OperatingSystem" is "Docker for Windows". I think that it is because it's actually the machine being run by Hyper-V, but I'm guessing. I didn't find any specific documentation about this case.